docs/Adding-Software-to-Homebrew.md

```diff
@@ -8,46 +8,139 @@ Before you start, please check the open pull requests for [Homebrew/homebrew-cor
 
 If you're stuck, ask for help on GitHub or the [Homebrew discussion forum](https://github.com/orgs/Homebrew/discussions). The maintainers are very happy to help but we also like to see that you've put effort into trying to find a solution first.
 
 ### Testing and auditing the formula
 
-1. Run `brew audit --strict --new-formula --online <formula>` with your formula. If any errors occur, correct your formula and run the audit again. The audit should finish without any errors by the end of this step.
+1. Test your formula installation using `brew install --formula --build-from-source $(brew --repository homebrew/core)/Formula/<prefix>/<formula>.rb`, where `<prefix>` is the first letter of your formula's name and `<formula>` is the name of your formula. Correct any errors that occur and repeat the installation until it completes without errors.
+
+1. Additionally, verify the proper uninstallation of the formula by using the command `brew uninstall --formula <formula>`, ensuring it removes the software without issues.
+
+1. Run `brew audit --formula --new <formula>` with your formula. If any errors occur, correct your formula and run the audit again. The audit should finish without any errors by the end of this step.
+
+1. Run `brew style --formula --fix <formula>` to automatically check and correct your formula's conformity to Homebrew's style guidelines.
 
 1. Run your formula's test using `brew test <formula>`. The test should finish without any errors.
 
+Keep in mind that these checks will happen automatically when you submit your pull request. Completing them beforehand not only saves time but also makes the whole process smoother for everyone.
+
+If your application and Homebrew do not work well together, feel free to [file an issue](https://github.com/Homebrew/homebrew-core/issues/new/choose) after checking out open issues.
+
 ### Submitting the formula
 
-You're finally ready to submit your formula to the [homebrew-core](https://github.com/Homebrew/homebrew-core) repository. If you haven't done this before, you can refer to the [How to Open a Homebrew Pull Request](How-To-Open-a-Homebrew-Pull-Request.md#formulae-related-pull-request) documentation for help. Maintainers will review the pull request and provide feedback about any areas that need to be addressed before the formula can be added to Homebrew.
+You're now ready to contribute your formula to the [homebrew-core](https://github.com/Homebrew/homebrew-core) repository. If this is your first time making a submission, consult the [How to Open a Homebrew Pull Request](How-To-Open-a-Homebrew-Pull-Request.md#formulae-related-pull-request) guide for detailed instructions. Your pull request will be reviewed by the maintainers, who may suggest improvements or changes before your formula can be added to Homebrew.
```

Comment:
Again: can you avoid changing wording here and stick to semantic changes in this PR, thanks.

Reply:
@MikeMcQuaid Original wording restored in 038035cf664b2b1e3b7ddeb2faacc76c3540023b, cask wording changed to align with that in 4a5d9cf5a1271b0b9951293dff7a1ba1a8dfafe3

I'll leave this comment thread open for visibility for now; but once you check it and are happy, leave a comment and then feel free to mark it resolved.